qemu/tests/timerdevice_change_guest_clksource.py

```diff
@@ -41,134 +32,41 @@ def verify_guest_clock_source(session, expected):
-                                                kernel_cfg_new)
-
-            error_context.context("Boot the guest", logging.info)
-            if clksrc != "kvm-clock":
-                cpu_model_flags = params.get("cpu_model_flags")
-                if "-kvmclock" not in cpu_model_flags:
-                    params["cpu_model_flags"] = cpu_model_flags + ",-kvmclock"
-            vm_name = params["main_vm"]
-            env_process.preprocess_vm(test, params, env, vm_name)
-            vm = env.get_vm(vm_name)
-            vm.verify_alive()
-            session = vm.wait_for_login(timeout=timeout)
-
-            error_context.context("Check the current clocksource in guest",
-                                  logging.info)
-            verify_guest_clock_source(session, clksrc)
-    finally:
-        try:
-            error_context.context("Shutdown guest")
-            vm.destroy()
-            error_context.context("Restore guest kernel cli", logging.info)
-            image_filename = storage.get_image_filename(params,
-                                                        data_dir.get_data_dir())
-            grub_file = params.get("grub_file", "/boot/grub2/grub.cfg")
-            kernel_cfg_pattern = params.get("kernel_cfg_pos_reg",
-                                            r".*vmlinuz-\d+.*")
-
-            disk_obj = utils_disk.GuestFSModiDisk(image_filename)
-            kernel_cfg_original = disk_obj.read_file(grub_file)
-            try:
-                kernel_cfg = re.findall(kernel_cfg_pattern,
-                                        kernel_cfg_original)[0]
-            except IndexError as detail:
-                test.error("Couldn't find the kernel config, regex"
-                           " pattern is '%s', detail: '%s'" %
-                           (kernel_cfg_pattern, detail))
-
-            if "clocksource=" in kernel_cfg:
-                kernel_cfg_new = re.sub(
-                    r"clocksource=[a-z \-_]+", " ", kernel_cfg)
-                disk_obj.replace_image_file_content(grub_file, kernel_cfg,
-                                                    kernel_cfg_new)
-        except Exception as detail:
-            logging.error("Failed to restore guest kernel cli."
-                          " Detail: '%s'" % detail)
+    for clksrc in available_clksrc_list:
+        if clksrc == "kvm-clock":
+            continue
+        error_context.context("Update guest kernel cli to '%s'" % clksrc,
+                              logging.info)
+        grubby_args = params["grubby_args"]
+        session.cmd(grubby_args % clksrc)
+        error_context.context("Reboot the guest", logging.info)
+        session = vm.reboot(session)
+        error_context.context("Check the current clocksource in guest",
+                              logging.info)
+        verify_guest_clock_source(session, clksrc)
+
+    error_context.context("Restore guest kernel cli", logging.info)
```

Comment:
If the test fails before this line, restoration of the clock source will never be executed, is this expected?